src/main/java/org/folio/service/transactions/batch/BatchEncumbranceService.java

```diff
@@ -72,23 +74,44 @@ private void updateBudget(Budget budget, CurrencyUnit currency, Transaction encu
         existingEncumbrance.getEncumbrance().getAmountAwaitingPayment(), currency);
       newAmount = subtractMoney(newAmount, existingEncumbrance.getEncumbrance().getAmountExpended(), currency);
       newAmount = sumMoney(newAmount, encumbrance.getEncumbrance().getAmountCredited(), currency);
+      // prevent the new encumbrance from exceeding the initial encumbered amount or from going negative
+      if (newAmount > encumbrance.getEncumbrance().getInitialAmountEncumbered()) {
+        newAmount = encumbrance.getEncumbrance().getInitialAmountEncumbered();
+      } else if (newAmount < 0) {
+        newAmount = 0;
```

Comment:
Is it added as safe point? or situations with 0 encumbrances are still possible?
in this case better to add warn log